dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java

```diff
@@ -940,14 +940,39 @@ public void adjustBundleBitstreamPolicies(Context context, Item item, Collection
 
             for (Bitstream bitstream : mybundle.getBitstreams()) {
                 // if come from InstallItem: remove all submission/workflow policies
-                authorizeService.removeAllPoliciesByDSOAndType(context, bitstream, ResourcePolicy.TYPE_SUBMISSION);
-                authorizeService.removeAllPoliciesByDSOAndType(context, bitstream, ResourcePolicy.TYPE_WORKFLOW);
-                addCustomPoliciesNotInPlace(context, bitstream, defaultItemPolicies);
-                addDefaultPoliciesNotInPlace(context, bitstream, defaultCollectionPolicies);
+                removeAllPoliciesAndAddDefault(context, bitstream, defaultItemPolicies, defaultCollectionPolicies);
             }
         }
     }
 
+    @Override
+    public void adjustBitstreamPolicies(Context context, Item item, Collection collection , Bitstream bitstream)
+        throws SQLException, AuthorizeException {
+        List<ResourcePolicy> defaultCollectionPolicies = authorizeService
+            .getPoliciesActionFilter(context, collection, Constants.DEFAULT_BITSTREAM_READ);
+
+        List<ResourcePolicy> defaultItemPolicies = authorizeService.findPoliciesByDSOAndType(context, item,
+                ResourcePolicy.TYPE_CUSTOM);
+        if (defaultCollectionPolicies.size() < 1) {
+            throw new SQLException("Collection " + collection.getID()
```

Comment:
This code is only the result of a refactoring so that we can apply the same logic to a single bitstream or to all the bitstreams in an item, see the existing method in the main branch
https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java#L922

Collections are required to have a DEFAULT_ITEM_READ and DEFAULT_BITSTREAM_READ policies, this is a constraint defined long time ago.